dask/array/tests/test_overlap.py

```diff
@@ -784,3 +784,22 @@ def test_ensure_minimum_chunksize_raises_error():
     chunks = (5, 2, 1, 1)
     with pytest.raises(ValueError, match="overlapping depth 10 is larger than"):
         ensure_minimum_chunksize(10, chunks)
+
+
+@pytest.mark.parametrize(
+    "shape, chunks, window_shape, axis",
+    [
+        ((3, 4, 5), (2, 3, 4), (3, 2), (1, 2)),
+        ((40, 30, 2), 5, (3,), (0,)),  # window < chunk
+        ((21,), 3, (7,), (0,)),  # window > chunk
```

Comment:
What about window == first chunk, since you had special-case code for that?

Other tests to consider:
- Chunk size varies along an axis (the first test partially covers that)
- `axis` is an integer.
- `axis` is None.
- `axis` contains the same axis several times.
- out-of-range axis
- negative axis (and same axis appearing several times, once specified as negative)
- zero-length window shape

Also some error cases to test:
- axis has different shape to window size
- window shape is larger than array shape on axis
- window shape has negative values

Reply:
very thorough list. Thank you.